neo4j/io/_bolt3.py

```diff
@@ -153,9 +155,10 @@ def hello(self):
             logged_headers["credentials"] = "*******"
         log.debug("[#%04X]  C: HELLO %r", self.local_port, logged_headers)
         self._append(b"\x01", (headers,),
-                     response=InitResponse(self, on_success=self.server_info._update_metadata))
+                     response=InitResponse(self, on_success=self.server_info.update))
         self.send_all()
         self.fetch_all()
+        check_supported_server_product(self.server_info.agent)
```

Comment:
This would raise if agent string changes. Why not just treat the agent string as whatever string?